### PR TITLE
Document NavigableElement

### PR DESCRIPTION
**Pull request by jknaudt21**
### Description of Change
Document `NavigableElement`. 

I also documented the top level summaries for `INavigation` and `NavigationProxy`. That being said, the internal implementations of these classes still needs to be documented. 

---

Part of documentation effort, related to https://github.com/dotnet/maui/issues/3960

### CODE REVIEW

File: src/Controls/src/Core/Shell/NavigableElement.cs [hunk 1]
```diff
@@ -63,13 +75,23 @@ public IList<string> @class
 
 		IList<string> IStyleSelectable.Classes => StyleClass;
 
-		/// <include file="../../../docs/Microsoft.Maui.Controls/NavigableElement.xml" path="//Member[@MemberName='NavigationProxy']/Docs/*" />
+		/// <summary>Gets the cast of <see cref="Navigation"/> to a <see cref="Maui.Controls.Internals.NavigationProxy"/>.</summary>
+		/// <remarks>
+		///		<para>Determines whether the element will proxy navigation calls</para>
```
Comment (davidbritch) [inline src/Controls/src/Core/Shell/NavigableElement.cs:80]: Missing period at the end of sentence.

**Review by davidbritch — DISMISSED** [review]
Approved, but one minor change required!

**Review by jsuarezruiz — DISMISSED** [review]

**Review by rmarinho — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
Test failures unrelated.

**Pull request merged**